chapters/hardware-software-interface/x86-architecture-family/reading/README.md

```diff
@@ -0,0 +1,118 @@
+# x86 Family
+
+Almost all major processors from Intel share a common ISA (Instruction Set Architecture).
+These processors are highly backward compatible, with most instructions unchanged over generations, but only added or extended.
+
+>**NOTE**: An ISA defines the instructions supported by a processor, register size, addressing modes, data types, instruction format, interrupts, and memory organization.
+>Processors in this family fall into the broad category of CISC (Complex Instruction Set Computers).
+>The philosophy behind them is to have a large number of instructions, with variable length, capable of performing complex operations, over multiple clock cycles.
+
+## Registers
+
+The basic working units for x86 processors are registers.
+These are a suite of locations within the processor through which it interacts with memory, I/O, etc.
+
+x86 processors have 8 such 32-bit registers.
+Although any of these can be used in operations, for historical reasons, each register has a specific role.
+
+Name | Role
+---- | ---
+EAX | accumulator; system calls, I/O, arithmetic
+EBX | base register; used for memory-based addressing
+ECX | counter in loop instructions
+EDX | data register, used for I/O, arithmetic, interrupt values; can extend EAX to 64 bits
+ESI | source in string operations
+EDI | destination in string operations
+EBP | base or frame pointer; points to the current stack frame
+ESP | stack pointer; points to the top of the stack
+
+In addition to these, there are some special registers that cannot be directly accessed by the programmer, such as `EFLAGS` and `eip` (Instruction Pointer).
```

Comment:
```suggestion
In addition to these, there are some special registers that cannot be directly accessed by the programmer, such as `eflags` and `eip` (Instruction Pointer).
```
Since `eflags` is also a register, we should use the same convention for it.